storage/innobase/log/log0recv.cc

```diff
@@ -3415,19 +3415,64 @@ recv_init_crash_recovery_spaces(bool rescan, bool& missing_tablespace)
 	return DB_SUCCESS;
 }
 
+dberr_t
+recv_recovery_read_max_checkpoint()
+{
+  ulint		max_cp_field;
+  dberr_t	err = DB_SUCCESS;
+
+  ut_ad(srv_operation <= SRV_OPERATION_EXPORT_RESTORED
+        || srv_operation == SRV_OPERATION_RESTORE
+        || srv_operation == SRV_OPERATION_RESTORE_EXPORT);
+  ut_d(mysql_mutex_lock(&buf_pool.flush_list_mutex));
+  ut_ad(UT_LIST_GET_LEN(buf_pool.LRU) == 0);
+  ut_ad(UT_LIST_GET_LEN(buf_pool.unzip_LRU) == 0);
+  ut_d(mysql_mutex_unlock(&buf_pool.flush_list_mutex));
+
+  if (srv_force_recovery >= SRV_FORCE_NO_LOG_REDO)
+  {
+    ib::info() << "innodb_force_recovery=6 skips redo log apply";
+    return DB_SUCCESS;
+  }
+
+  mysql_mutex_lock(&log_sys.mutex);
+
+  err= recv_find_max_checkpoint(&max_cp_field);
+
+  if (err != DB_SUCCESS)
+  {
+     recv_sys.recovered_lsn = log_sys.get_lsn();
+     mysql_mutex_unlock(&log_sys.mutex);
+     return err;
+  }
+
+  byte* buf= log_sys.checkpoint_buf;
+  if ((err= log_sys.log.read(max_cp_field,
+		{buf, OS_FILE_LOG_BLOCK_SIZE})))
+  {
+    mysql_mutex_unlock(&log_sys.mutex);
+    return err;
+  }
+
+  log_sys.next_checkpoint_no= mach_read_from_8(
+				buf + LOG_CHECKPOINT_NO);
+  log_sys.next_checkpoint_lsn= mach_read_from_8(
+				buf + LOG_CHECKPOINT_LSN);
+  recv_sys.mlog_checkpoint_lsn= mach_read_from_8(
+				buf + LOG_CHECKPOINT_END_LSN);
+  mysql_mutex_unlock(&log_sys.mutex);
```

Comment:
If we have to split lines, we should avoid it right after `(`. Instead, split the line after the assignment operator.

Can we avoid duplicating the `mysql_mutex_unlock()` calls?
```c++
  mysql_mutex_lock(&log_sys.mutex);
  ulint max_cp;
  dberr_t err= recv_find_max_checkpoint(&max_cp);

  if (err != DB_SUCCESS)
    recv_sys.recovered_lsn= log_sys.get_lsn();
  else
  {
    byte *buf= log_sys.checkpoint_buf;
    err= log_sys.log.read(max_cp, {buf, OS_FILE_LOG_BLOCK_SIZE});
    if (err == DB_SUCCESS)
    {
      log_sys.next_checkpoint_no= mach_read_from_8(buf + LOG_CHECKPOINT_NO);
      log_sys.next_checkpoint_lsn= mach_read_from_8(buf + LOG_CHECKPOINT_LSN);
      recv_sys.mlog_checkpoint_lsn=
        mach_read_from_8(buf + LOG_CHECKPOINT_END_LSN);
    }
  }

  mysql_mutex_unlock(&log_sys.mutex);
  return err;